DataPlotly/gui/plot_settings_widget.py

```diff
@@ -1390,7 +1390,10 @@ def read_project(self, document: QDomDocument):
         settings = PlotSettings()
         if settings.read_from_project(document):
             # update the dock state to match the read settings
-            self.set_settings(settings)
+            try:
+                self.set_settings(settings)
```

Comment:
I think we'd need to address this on a case-by-case basis within set_settings. Otherwise we have the situation that a newly added setting will potentially cause other existing settings to be skipped (when set_settings encounters a missing key **before** other settings which do exist are processed)

Reply:
mmm ok got it. So a try - except for every key within the `set_settings` method?

Reply:
I have used the get method. It allows to set a default value. e.g.

`settings.layout.get('x_min', None)`

(this reads the layout property 'x_min', if it is not available, None is used)

Reply:
Yep, get would be the best approach here

Reply:
`get` approach used. If OK I'm going to merge it!